src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoroelasticKernel.hpp

```diff
@@ -0,0 +1,447 @@
+  using Base::m_elemsToNodes;
+  using Base::m_constitutiveUpdate;
+  using Base::m_finiteElementSpace;
+
+
+  /**
+   * @brief Constructor
+   * @copydoc geosx::finiteElement::ImplicitKernelBase::ImplicitKernelBase
+   * @param inputGravityVector The gravity vector.
+   */
+  SinglePhase( NodeManager const & nodeManager,
+               EdgeManager const & edgeManager,
+               FaceManager const & faceManager,
+               localIndex const targetRegionIndex,
+               SUBREGION_TYPE const & elementSubRegion,
+               FE_TYPE const & finiteElementSpace,
+               CONSTITUTIVE_TYPE & inputConstitutiveType,
+               arrayView1d< globalIndex const > const & inputDispDofNumber,
+               string const & inputFlowDofKey,
+               globalIndex const rankOffset,
+               CRSMatrixView< real64, globalIndex const > const & inputMatrix,
+               arrayView1d< real64 > const & inputRhs,
+               real64 const (&inputGravityVector)[3],
+               arrayView1d< string const > const & fluidModelNames ):
+    Base( nodeManager,
+          edgeManager,
+          faceManager,
+          targetRegionIndex,
+          elementSubRegion,
+          finiteElementSpace,
+          inputConstitutiveType,
+          inputDispDofNumber,
+          rankOffset,
+          inputMatrix,
+          inputRhs ),
+    m_X( nodeManager.referencePosition()),
+    m_disp( nodeManager.totalDisplacement()),
+    m_uhat( nodeManager.incrementalDisplacement()),
+    m_gravityVector{ inputGravityVector[0], inputGravityVector[1], inputGravityVector[2] },
+    m_gravityAcceleration( sqrt( inputGravityVector[0] * inputGravityVector[0] +
+                                 inputGravityVector[1] * inputGravityVector[1] +
+                                 inputGravityVector[2] * inputGravityVector[2] ) ),
+    m_solidDensity( inputConstitutiveType.getDensity() ),
+    m_fluidDensity( elementSubRegion.template getConstitutiveModel< constitutive::SingleFluidBase >( fluidModelNames[targetRegionIndex] ).density() ),
+    m_fluidDensityOld( elementSubRegion.template getReference< array1d< real64 > >( SinglePhaseBase::viewKeyStruct::densityOldString() ) ),
+    m_dFluidDensity_dPressure( elementSubRegion.template getConstitutiveModel< constitutive::SingleFluidBase >( fluidModelNames[targetRegionIndex] ).dDensity_dPressure() ),
+    m_flowDofNumber( elementSubRegion.template getReference< array1d< globalIndex > >( inputFlowDofKey )),
+    m_fluidPressure( elementSubRegion.template getReference< array1d< real64 > >( SinglePhaseBase::viewKeyStruct::pressureString() ) ),
+    m_deltaFluidPressure( elementSubRegion.template getReference< array1d< real64 > >( SinglePhaseBase::viewKeyStruct::deltaPressureString() ) ),
+    m_poroRef( elementSubRegion.template getReference< array1d< real64 > >( SinglePhaseBase::viewKeyStruct::referencePorosityString() ) ),
+    m_biotCoefficient( m_constitutiveUpdate.getBiotCoefficient() )
```

Comment:
Isotropic Biot's tensor is considered here (single Biot's coefficient), so this class should be called IsotropicSinglePhase? Do we pretend to implement anisotropic Biot's tensor?

Reply:
Appropriate Biot's coefficient, porosity, etc. constitutive models will be added in future PRs.